new_platforms/scripts/VirtualMachineSgxSettings.psm1

```diff
@@ -0,0 +1,379 @@
+        SGX EPC memory size to 32M, do:
+
+        Set-VMSgx -VmName MySgxVM -IsSgxEnabled $True -SgxSize 32
+
+        You may also write:
+
+        Set-VMSgx MySgxVM $True 32
+
+    .Example
+        To disable SGX virtualization on a virtual machine by name, do:
+
+        Set-VMSgx -VmName MySgxVM -IsSgxEnabled $False -SgxSize 0
+
+        Alternately:
+
+        Set-VMSgx MySgxVM $False 0
+
+    .Link
+        https://openenclave.io/
+#>
+Function Set-VMSgx {
+    [CmdletBinding()]
+    Param
+    (
+        # Name of the VM to configure.
+        [Parameter(Mandatory = $True, ParameterSetName = "ByName", Position = 0)]
+        [ValidateNotNullOrEmpty()]
+        [String]$VmName,
+
+        # VM to configure.
+        [Parameter(Mandatory = $True, ParameterSetName = "ByObject", Position = 0)]
+        [ValidateNotNull()]
+        [Microsoft.HyperV.PowerShell.VirtualMachine]$Vm,
+
+        # Whether SGX is enabled in the VM.
+        [Parameter(Position = 1)]
+        [Bool]$IsSgxEnabled = $False,
+
+        # Desired SGX EPC Memory Size for the VM (in MB).
+        [Parameter(Position = 2)]
+        [UInt64]$SgxSize = 0,
+
+        # Desired default SGX Launch Control Mode for the VM.
+        [Parameter(Position = 3)]
+        [String]$SgxLaunchControlDefault = [String]::Empty,
+
+        # Desired SGX Launch Control Mode for the VM.
+        [Parameter(Position = 4)]
+        [UInt32]$SgxLaunchControlMode = 0,
+
+        # Ignore warnings.
+        [Parameter()]
+        [Switch]$Force
+    )
+
+    Process {
+        # If the user passed a VM name, retrieve the VM object.
+        If ($Null -eq $Vm) {
+            $Vm = Get-VM $VmName
+        }
+
+        # If the VM is not Off, the function cannot proceed.
+        If ($Vm.State -ne [Microsoft.HyperV.PowerShell.VMState]::Off) {
+            Write-Error ($Strings.ERR_VM_STATE -f $Vm.Name)
+        }
+
+        # At the time of this writing, the following conditions hold:
+        #
+        # 1. SGX virtualization is only supported on Gen2 VMs:
+        # 1.1. The Intel SGX driver is not yet merged into mainline Linux and must be compiled manually;
+        #      For VMs with Secure Boot on, this means that the module must be signed with a custom key;
+        #      Note: Only Gen2 VMs support Secure Boot.
+        # 2. Checkpoints are not supported with SGX turned on.
+
+        If (!$Force -and $IsSgxEnabled) {
+            # Check VM generation.
+            If ($Vm.Generation -lt 2) {
+                Write-Warning ($Strings.WARN_GEN2_TITLE -f $Vm.Name)
+                Write-Warning $Strings.WARN_GEN2_TEXT
+            }
+
+            # Check Secure Boot settings.
+            If ($Vm.Generation -ge 2) {
+                $Fw = $Vm | Get-VMFirmware
+
+                If ($Fw.SecureBoot -eq [Microsoft.HyperV.PowerShell.OnOffState]::On) {
+                    Write-Information ($Strings.WARN_SECURE_BOOT_TITLE -f $Vm.Name)
```

Comment:
Are you aware of this?

The $InformationPreference preference variable value determines whether the message you provide to Write-Information is displayed at the expected point in a script's operation. Because the default value of this variable is SilentlyContinue, by default, informational messages are not shown. 

https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/write-information?view=powershell-6

Reply:
Yes, I noticed when the messages would not print.

The value of `$InformationPreference` is set at the top of the module.